### ISSUE TITLE
Remove slf4j from the dependencies

### ISSUE THREAD

**Issue by Frontrider** [javadelight/delight-nashorn-sandbox#48]
I had some trouble integrating it because of this dependency, and it doesn't really serves any vital purpose.
What it is actually used for, can be substituted with a the default out/err methods, and a boolean flag.

Would make the library more light weight and easier to integrate, with existing systems.

It's a good thing to have a logging library, but here it feels a little overkill.

**Comment by mxro** [comment]
Personally I am not opposed to removing the logging dependency. For my own projects, I like to use this implementation [delight-simple-log](https://github.com/javadelight/delight-simple-log). Very lightweight and should be easy to integrate with any environment since the dependency to log4j /slf4j is optional.

What were your specific problems in integrating the library in its current form?

**Comment by Frontrider** [comment]
I'll say the usecase:
I'm experimenting with a minecraft mod, which adds computers what you can program using javascript. This library, is a really good thing here, since I do not need to bother making the execution secure. Nobody wants to have their server get into an endless cycle of shutdown by a troll calling exit, on a ticking event or something.

MC already has a logging library in it, and its a good practice to remove useless dependencies anyways. It'll create less bloat.

**Comment by Frontrider** [comment]
At this point, I'll just ship my own version.

**Comment by cmorriss** [comment]
I'll put in my 2cents. The point of Slf4j is to provide a single interface that has many different implementations. It's actually great to use in a library as it allows the library user to forward the information to their logs very easily. If you were to switch to System.out and System.err, it would make this more difficult as lots of information can be sent to those streams and you may want to more tightly control the amount and direction.

Net of it is, I say stick with Slf4j.

**Comment by mrmakeit** [comment]
I can't tell if @Frontrider is referring to [OCJS](https://github.com/mrmakeit/OCJS). If not I'd like to point out that I am having the same issue when trying to integrate this library as well.  I've tried to embed the slf4j dependency with no effect, and can't shade it without rebuilding the entire delight-nashorn-sandbox project.  I'm excited to try and get OCJS running, so I'm open to suggestions.

**Comment by Frontrider** [comment]
Integration wise, I'd have a kind of solution. Can't do it now, but I'd still like to throw it out there.
Make it log into an `OutputStream` or similar. Than anyone can take it however they want it.

Also, I take note of that mod. One project where I wanted to integrate this library, was actually a computer mod, and your take is better.

**Comment by mxro** [comment]
Hello Everyone! Good to see this issue is continued to be debated!

So @mrmakeit just to confirm, where you able to get a workaround working?

**Comment by Frontrider** [comment]
And @mrmakeit , no I'm not reffering to OCJS. I tried to make my own version with slightly different principles, but than I failed while integrating a few things (strictly on my end).

**Comment by Frontrider** [comment]
That was completly unintended

**Comment by mrmakeit** [comment]
@mxro yes, my current for builds and seems to work for now.  I can submit a pr if you'd like.
@Frontrider sorry to hear it didn't work out.  Your more than welcome to come help out, as there are still some big gaps to be fixed.

**Comment by mxro** [comment]
Clearing up old issues - if this issue still persists, please be welcome to reopen this one or open a new one!
